### PR TITLE
Remove optional transitive `tink` and `protobuf-java` dependencies

### PR DESCRIPTION
**Pull request by slobodanadamovic**
This commit removes `com.google.crypto.tink` which is transitive and optional dependency of `oauth2-oidc-sdk` and `nimbus-jose-jwt`. We don't seem to be using any functionality that requires `tink` and thus `protobuf-java`. Removing them feels safer than having to maintain misaligned versions.

Note: Marking this as `non-issue` since the https://github.com/elastic/elasticsearch/pull/111344 PR (that introduced them) hasn't been released yet.

### CODE REVIEW

File: modules/repository-azure/build.gradle [hunk 1]
```diff
@@ -63,8 +63,12 @@ dependencies {
   api "com.github.stephenc.jcip:jcip-annotations:1.0-1"
   api "com.nimbusds:content-type:2.3"
   api "com.nimbusds:lang-tag:1.7"
-  api "com.nimbusds:nimbus-jose-jwt:9.37.3"
-  api "com.nimbusds:oauth2-oidc-sdk:11.9.1"
+  api("com.nimbusds:nimbus-jose-jwt:9.37.3"){
+    exclude group: 'com.google.crypto.tink', module: 'tink' // it's an optional dependency to which we don't rely on
```
Comment (DaveCTurner) [inline modules/repository-azure/build.gradle:67]: grammar nit 🙂 

```suggestion
    exclude group: 'com.google.crypto.tink', module: 'tink' // it's an optional dependency on which we don't rely
```

**Review by DaveCTurner — APPROVED** [review]
LGTM

File: modules/repository-azure/build.gradle (hunk 1 above)
Comment (slobodanadamovic) [reply modules/repository-azure/build.gradle]: 👍 pushed https://github.com/elastic/elasticsearch/pull/115916/commits/bdd243d4a82b78f15c15e06a44825fa32ea5e5b1

**Comment by slobodanadamovic** [comment]
`elasticsearch-ci/8.17.0 / bwc-snapshots` failure is unrelated. I will retry that one 🤞

**Comment by slobodanadamovic** [comment]
@elasticmachine update branch

**Comment by slobodanadamovic** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/elasticsearch/pull/115953)|
|✅|8.16|[](https://github.com/elastic/elasticsearch/pull/115954)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**